src/test/java/com/sixt/service/framework/rpc/HttpClientWrapperTest.java

```diff
@@ -0,0 +1,119 @@
+package com.sixt.service.framework.rpc;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyList;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.sixt.service.framework.OrangeContext;
+import com.sixt.service.framework.ServiceProperties;
+import com.sixt.service.framework.metrics.GoTimer;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.HttpContentResponse;
+import org.eclipse.jetty.client.api.ContentProvider;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.Date;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+
+public class HttpClientWrapperTest {
+
+    private static final int NUMBER_OF_RETRIES = 3;
+
+    private ServiceProperties serviceProperties = mock(ServiceProperties.class);
+    private HttpClient httpClient = mock(HttpClient.class);
+    private RpcClientMetrics rpcClientMetrics = mock(RpcClientMetrics.class);
+    private Tracer tracer = mock(Tracer.class);
+    private RpcClient rpcClient = mock(RpcClient.class);
+    private LoadBalancer loadBalancer = mock(LoadBalancer.class);
+    private Request request = mock(Request.class);
+
+    private Tracer.SpanBuilder spanBuilder = mock(Tracer.SpanBuilder.class);
+    private Span span = mock(Span.class);
+
+    private HttpContentResponse httpContentResponse = mock(HttpContentResponse.class);
+    private JsonRpcCallExceptionDecoder decoder = mock(JsonRpcCallExceptionDecoder.class);
+
+    private HttpClientWrapper httpClientWrapper
+        = new HttpClientWrapper(serviceProperties, httpClient, rpcClientMetrics, tracer);
+
+    @Before
+    public void setup() throws InterruptedException, ExecutionException, TimeoutException {
+        when(loadBalancer.getHealthyInstance()).thenReturn(createServiceEndpoint());
+        when(loadBalancer.getHealthyInstanceExclude(anyList())).thenReturn(createServiceEndpoint());
```

Comment:
you could use `anyListOf(ServiceEndpoint.class)` instead of `anyList()` in order to have a checked assignment here.